tools/buildmgr/cbuild/include/CbuildModel.h

```diff
@@ -508,19 +508,19 @@ class CbuildModel {
   std::map<std::string, std::set<std::string>>      m_layerPackages;
 
   void Init(const std::string &file, const std::string &rtePath);
-  bool EvaluateResult();
-  bool EvalConfigFiles();
-  bool EvalPreIncludeFiles();
-  bool EvalSourceFiles();
+  bool EvaluateResult(bool checkFileExistence);
+  bool EvalConfigFiles(bool checkFileExistence);
+  bool EvalPreIncludeFiles(bool checkFileExistence);
+  bool EvalSourceFiles(bool checkFileExistence);
   bool EvalNonRteSourceFiles();
   bool EvalGeneratedSourceFiles();
-  bool EvalRteSourceFiles(std::map<std::string, std::list<std::string>> &cSourceFiles, std::map<std::string, std::list<std::string>> &cxxSourceFiles, std::map<std::string, std::list<std::string>> &asmSourceFiles, std::string &linkerScript);
-  bool EvalFile(RteItem* file, const std::string& group, const std::string& base, std::string& filepath);
-  bool EvalItem(RteItem* item, const std::string& groupName = std::string(), const std::string& groupLayer = std::string());
+  bool EvalRteSourceFiles(std::map<std::string, std::list<std::string>> &cSourceFiles, std::map<std::string, std::list<std::string>> &cxxSourceFiles, std::map<std::string, std::list<std::string>> &asmSourceFiles, std::string &linkerScript, bool checkFileExistence);
+  bool EvalFile(RteItem* file, bool checkFileExistence, const std::string& group, const std::string& base, std::string& filepath);
+  bool EvalItem(RteItem* item, bool checkFileExistence, const std::string& groupName = std::string(), const std::string& groupLayer = std::string());
   bool EvalItemTranslationControls(const RteItem* item, TranslationControlsKind kind, const std::string& groupName = std::string());
   bool GenerateRteHeaders();
   bool EvalDeviceName();
-  bool EvalFlags();
+  bool EvalFlags(bool checkFileExistence);
```

Comment:
To avoid the propagation of the `checkFileExistence` argument over all these functions, would it be possible to have a `CbuildModel` class member variable instead?